StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowController.swift

```diff
@@ -114,18 +114,30 @@ extension NativeFlowController {
         navigationController.setViewControllers(viewControllers, animated: animated)
     }
 
-    private func pushPane(_ pane: FinancialConnectionsSessionManifest.NextPane, animated: Bool) {
+    private func pushPane(
+        _ pane: FinancialConnectionsSessionManifest.NextPane,
+        animated: Bool,
+        // useful for cases where we want to prevent the user from navigating back
+        //
+        // keeping this logic in `pushPane` is helpful because we want to
+        // reuse `skipSuccessPane` and `manualEntryMode == .custom` logic
+        clearNavigationStack: Bool = false
+    ) {
         if pane == .success && dataManager.manifest.skipSuccessPane == true {
             closeAuthFlow(showConfirmationAlert: false, error: nil)
         } else if pane == .manualEntry && dataManager.manifest.manualEntryMode == .custom {
             closeAuthFlow(showConfirmationAlert: false, customManualEntry: true)
         } else {
-            let manualEntryViewController = CreatePaneViewController(
+            let paneViewController = CreatePaneViewController(
                 pane: pane,
                 nativeFlowController: self,
                 dataManager: dataManager
             )
-            pushViewController(manualEntryViewController, animated: animated)
+            if clearNavigationStack, let paneViewController = paneViewController {
```

Comment:
because `clearNavigationStack` default value is set to `false`, this should mean that the changes to this function are a no-op for old callers